### PR TITLE
docs(DEVELOPER.md): explain how to get the Rust build system authenticated by GitHub

### PR DESCRIPTION
**Pull request by ADD-SP**
### Summary

* Added two new ways to do authentication with GitHub.
* Explained how to get the Rust build system authenticated by GitHub.

### Checklist

- [N/A] The Pull Request has tests
- [N/A] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [N/A] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### CODE REVIEW

**Review by Water-Melon — APPROVED** [review]

File: DEVELOPER.md [hunk 1]
```diff
@@ -138,11 +138,19 @@ macOS
 brew install libyaml
 ```
 
-Now, we have to set environment variable `GITHUB_TOKEN` to download some essential repos.
-You can follow [Managing your personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) to generate an access token. It does not need to have any other permission than `Public Repositories (read-only)`.
+Now, you have to authenticate with GitHub to download some essential repos.
+There are three ways to do it:
+* By [`gh`](https://cli.github.com/), basically `gh auth login`.
```
Comment (fffonion) [inline DEVELOPER.md:143]:
```suggestion
* Download [`gh cli`](https://cli.github.com/) and run `gh auth login` once.
```

File: DEVELOPER.md [hunk 2]
```diff
@@ -404,24 +412,24 @@ To avoid long compilation times, give the VM plenty of RAM (8GB recommended) and
...
 1. Select the virtual machine you want to use and click "Settings"
-1. Click the "Network" tab
-1. Click the "Advanced" dropdown
-1. Click "Port Forwarding"
-1. Add a new rule in the popup. The only thing you will need is "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)
-1. Click "Ok"
+2. Click the "Network" tab
+3. Click the "Advanced" dropdown
+4. Click "Port Forwarding"
+5. Add a new rule in the popup. The only thing you will need is "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)
```
Comment (ProBrian) [inline DEVELOPER.md:418]:
```suggestion
5. Add a new rule in the popup. The only thing you will need is set "Host Port" to be 22222 and "Guest Port" to be 22. Everything else can be left default (see screenshot below)
```

File: DEVELOPER.md [hunk 3]
```diff
@@ -138,11 +138,19 @@ macOS
 brew install libyaml
 ```
 
-Now, we have to set environment variable `GITHUB_TOKEN` to download some essential repos.
-You can follow [Managing your personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) to generate an access token. It does not need to have any other permission than `Public Repositories (read-only)`.
+Now, you have to authenticate with GitHub to download some essential repos.
+There are three ways to do it:
```
Comment (fffonion) [inline DEVELOPER.md:142]:
```suggestion
Now, you have to authenticate with GitHub to download some essential repos
using either one of the following ways:
```

File: DEVELOPER.md [hunk 4]
```diff
@@ -138,11 +138,19 @@ macOS
...
+* By [`gh`](https://cli.github.com/), basically `gh auth login`.
+* By [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). This token does not need to have any other permission than `Public Repositories (read-only)`, and set it as `GITHUB_TOKEN` environment variable.
```
Comment (fffonion) [inline DEVELOPER.md:144]:
```suggestion
* Use a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). This token does not need to have any other permission than `Public Repositories (read-only)`, and set it as `GITHUB_TOKEN` environment variable.
```

**Review by ProBrian — APPROVED** [review]

File: DEVELOPER.md [hunk 5]
```diff
@@ -138,11 +138,19 @@ macOS
...
+* By [git credential helper](https://git-scm.com/docs/gitcredentials).
```
Comment (fffonion) [inline DEVELOPER.md:145]:
```suggestion
* Use [git credential helper](https://git-scm.com/docs/gitcredentials).
```

File: DEVELOPER.md [hunk 6]
```diff
@@ -138,11 +138,19 @@ macOS
...
 
-```bash
-# export GITHUB_TOKEN=ghp_xxxxxx_your_access_token
+Then you have to make the rust build system also authenticate with GitHub,
```
Comment (dndx) [inline DEVELOPER.md:147]:
```suggestion
Then you have to make the Rust build system also authenticate with GitHub,
```

File: DEVELOPER.md (hunk 2 above)
Comment (dndx) [reply DEVELOPER.md:418]: You can use all `1.`s in Markdown and let the generator automatically maintain the sequence, it is a common practice and makes adding additional items easier.

File: DEVELOPER.md [hunk 7]
```diff
@@ -404,24 +412,24 @@ To avoid long compilation times, give the VM plenty of RAM (8GB recommended) and
...
-1. Logout of guest and make sure you are not promoted password when SSH again
-1. Edit the .ssh/config file on your host and put in the following content:
+2. Type `cat .ssh/id_ed25519.pub` and copy the public key
+3. SSH into the guest using the command above
+4. Create the ssh config directory (if it doesn't exist) `$ mkdir -p .ssh`
+5. Edit the authorized keys list: `vim .ssh/authorized_keys`
+6. Paste in the content of .ssh/id_ed25519.pub
+7. Adjust the required privileges: `chmod 700 .ssh/`  and `chmod 400 .ssh/authorized_keys`
+8. Logout of guest and make sure you are not promoted password when SSH again
+9. Edit the .ssh/config file on your host and put in the following content:
```
Comment (dndx) [inline DEVELOPER.md:432]: Again, nothing to worry about leaving it all `1`s.

Comment (ADD-SP) [reply DEVELOPER.md]: reverted

File: DEVELOPER.md (hunk 2 above)
Comment (ADD-SP) [reply DEVELOPER.md]: reverted

**Review by oowl — APPROVED** [review]

**Comment by ADD-SP** [comment]
Manually cherry-picked: https://github.com/Kong/kong-ee/pull/9894

**Pull request merged**